pipelines/SARS-CoV-2-pipelines/run_python_scripts/rules/compare_mutations.py

```diff
@@ -0,0 +1,148 @@
+    parser.add_argument("file_path", type=str)
+    parser.add_argument("reference", type=str)
+    parser.add_argument("mutations", type=str)
+    parser.add_argument("--threshold", type=int, default=42) #minimal number of reads needed
+    parser.add_argument("-o", "--output", type=str, required=True)
+    return parser.parse_args(argv)
+    
+def load_file(file_path, reference):
+    barcode_dict = {}
+    with open(file_path, "r") as f:
+        for line in f:
+            l = line.split(",")
+            position = int(l[0])-1
+            barcode = l[1]
+            A = int(l[2])
+            C = int(l[3])
+            G = int(l[4])
+            T = int(l[5])
+            if not barcode in barcode_dict:
+                barcode_dict[barcode] = [[0 for _ in letters] for _ in reference]
+            barcode_dict[barcode][position][0]+=A
+            barcode_dict[barcode][position][1]+=C
+            barcode_dict[barcode][position][2]+=G
+            barcode_dict[barcode][position][3]+=T   
+    return barcode_dict
+
+
+def load_mutations(mutations_path):    #loading .txt file containing mutations for each variant
+    
+    mutations = Tree("root", 0)
+    stack = []
+    index = 0
+    stack.append(mutations)
+    
+    with open (mutations_path, "r") as txtfile:
+        mut_name = ""
+        maybeparent = mutations
+        for line in txtfile:
+            line = line.strip()
+            if line == "start_sub":
+                stack.append(maybeparent)
+                index += 1
+            elif line == "end_sub":
+                stack.pop()   
+                index -= 1       
+            elif not line.startswith('#') and line:
+                columns = line.split(" ")
+                parent = stack[index]  
+                child = Tree(columns[0], int(columns[1]))
+                for i in range(2, len(columns)):
+                    child.data.append(columns[i])
+		
+                parent.children.append(child)
+                maybeparent = child            
+
+    return mutations
+
+def find_variants(barcode_dict, variants, json, threshold): #looking for possible variants for a barcode
+    barcode_strings=[]
+    json_string='['
+    for barcode in barcode_dict:
+        barcode_string=""
+        barcode_string+='{"barcode":"'+barcode+'","variants":[' #begin of barcode
+        variant_strings=[]
+        for strand in variants.children:
+            variant_string=check_variant(barcode_dict[barcode], threshold, strand)
+            if variant_string!="":
+                variant_strings.append(variant_string)
+        barcode_string+= ", ".join(variant_strings)
+        barcode_string+=']}' #end of barcode
+        barcode_strings.append(barcode_string)
+    json_string+=", ".join(barcode_strings)
+    json_string+=']'                        
+    print(json_string, file=json)      
+
+
+def check_variant(barcode, threshold, strand): #check whether this variant meets our conditions
+    pocet=0
+    json_string = '{'
```

Comment:
why do you build json string instead of building dict and then using `json.dumps()` ?

Reply:
because I did not know of the `json.dumps()` method. I should have googled it in the first place..  Thanks. I'll use it.